examples/neural_networks/plot_random_neural_network.py

```diff
@@ -0,0 +1,155 @@
+
+    X += 2 * rng.uniform(size=X.shape)
+    linearly_separable = (X, y)
+
+    datasets = [make_moons(noise=0.3, random_state=rng),
+                make_circles(noise=0.2, factor=0.5, random_state=rng),
+                linearly_separable]
+
+    figure = plt.figure(figsize=(17, 9))
+    figure.suptitle(title)
+    i = 1
+    # iterate over datasets
+    for X, y in datasets:
+        # initialize standard scaler
+        scaler = StandardScaler()
+
+        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=.4,
+                                                            random_state=1)
+        # Compute the mean and standard deviation of each feature of the
+        # training set and scale the training set
+        X_train = scaler.fit_transform(X_train)
+
+        # Using the same mean and standard deviation, scale the testing set
+        X_test = scaler.transform(X_test)
+
+        x_min, x_max = X[:, 0].min() - .5, X[:, 0].max() + .5
+        y_min, y_max = X[:, 1].min() - .5, X[:, 1].max() + .5
+        xx, yy = np.meshgrid(np.arange(x_min, x_max, h),
+                             np.arange(y_min, y_max, h))
+
+        # just plot the dataset first
+        cm_bright = ListedColormap(['#FF0000', '#0000FF'])
+        ax = plt.subplot(len(datasets), len(classifiers) + 1, i)
+        # Plot the training points
+        ax.scatter(X_train[:, 0], X_train[:, 1], c=y_train, cmap=cm_bright)
+        # and testing points
+        ax.scatter(X_test[:, 0], X_test[:, 1], c=y_test, cmap=cm_bright,
+                   alpha=0.6)
+        ax.set_xlim(xx.min(), xx.max())
+        ax.set_ylim(yy.min(), yy.max())
+        ax.set_xticks(())
+        ax.set_yticks(())
+        i += 1
+
+        # iterate over classifiers
+        for name, clf in zip(names, classifiers):
+            ax = plt.subplot(len(datasets), len(classifiers) + 1, i)
+            clf.fit(X_train, y_train)
+            score = clf.score(X_test, y_test)
+
+            # Plot the decision boundary.
+            Z = predict_proba(clf, np.c_[xx.ravel(), yy.ravel()])
+
+            # Put the result into a color plot
+            Z = Z.reshape(xx.shape)
+
+            ax.contourf(xx, yy, Z, cmap=plt.cm.RdBu, alpha=.8)
+
+            # Plot also the training points
+            ax.scatter(X_train[:, 0], X_train[:, 1], c=y_train, cmap=cm_bright)
+            # and testing points
+            ax.scatter(X_test[:, 0], X_test[:, 1], c=y_test, cmap=cm_bright,
+                       alpha=0.6)
+
+            ax.set_xlim(xx.min(), xx.max())
+            ax.set_ylim(yy.min(), yy.max())
+            ax.set_xticks(())
+            ax.set_yticks(())
+            ax.set_title(name)
+            ax.text(xx.max() - .3, yy.min() + .3, ('%.2f' % score).lstrip('0'),
+                    size=15, horizontalalignment='right')
+            i += 1
+
+classifiers = []
+names = []
+for alpha in alpha_list:
+    clf = make_pipeline(RandomBasisFunction(weight_scale=1.), Ridge(alpha=1.))
```

Comment:
`validation_curve` seems more tricky to use here since I am plotting the decision function for each `alpha` and accumulating the classifiers with different parameters in a list.